fpdf/fpdf.py

```diff
@@ -2072,56 +2081,84 @@ def rotation(self, angle, x=None, y=None):
     @contextmanager
     def local_context(
         self,
-        draw_color=None,
-        fill_color=None,
-        text_color=None,
         font_family=None,
         font_style=None,
         font_size=None,
+        line_width=None,
+        draw_color=None,
+        fill_color=None,
+        text_color=None,
+        dash_pattern=None,
         **kwargs,
     ):
         """
-        Create a local grapics state, which won't affect the surrounding code.
+        Creates a local graphics state, which won't affect the surrounding code.
         This method must be used as a context manager using `with`:
 
             with pdf.local_context():
                 set_some_state()
                 draw_some_stuff()
 
-        The affected settings are all those controlled by the GraphicsStateMixin:
+        The affected settings are those controlled by GraphicsStateMixin and drawing.GraphicsStyle:
+            allow_transparency
+            auto_close
+            blend_mode
+            dash_pattern
             draw_color
             fill_color
```

Comment:
>  It looks like in the current implementation you can only set the former through the argument list.

That's perfectly right.

>  Should they be synchronized somehow? Or am I missing any special magic here?

No, I decided that for now `local_context()` will only support `fill_color` as a PDF stream operator.
It was not needed to set it in the `GraphicsStyle` object, so I did bother to do so.
This could be improved on later, if need be.